python/pyspark/sql/functions.py

```diff
@@ -969,8 +1005,8 @@ def months_between(date1, date2):
     """
     Returns the number of months between date1 and date2.
 
-    >>> df = spark.createDataFrame([('1997-02-28 10:30:00', '1996-10-30')], ['t', 'd'])
-    >>> df.select(months_between(df.t, df.d).alias('months')).collect()
+    >>> df = spark.createDataFrame([('1997-02-28 10:30:00', '1996-10-30')], ['date1', 'date2'])
```

Comment:
I reverted all the other ones, but I like this one because `date1` and `date2` match the docstring. Is that reasonable?